recirq/qcqmc/fermion_mode.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import attrs
+
+
+@attrs.frozen
+class FermionicMode:
+    """A specification of a fermionic mode.
+
+    Args:
+        orb_ind: The spatial orbital index.
+        spin: The spin state of the fermion mode (up or down (alpha or beta)).
```

Comment:
specify that it should be specifically the string "a" or "b" (I think doug called for an enum before -- I think a string and the validator is fine for this use case)

Reply:
I tried enum but then ran into a serialization issue and gave up so it seemed like more trouble than it was worth. I'll add a comment.

Reply:
Done.